exchanges/kucoin/kucoin_test.go

```diff
@@ -0,0 +1,2419 @@
+	}
+
+	_, err := ku.PostBorrowOrder(context.Background(), "USDT", "FOK", "", 10, 0)
+	if err != nil {
+		t.Error("PostBorrowOrder() error", err)
+	}
+
+	_, err = ku.PostBorrowOrder(context.Background(), "USDT", "IOC", "7,14,28", 10, 10)
+	if err != nil {
+		t.Error("PostBorrowOrder() error", err)
+	}
+}
+
+func TestGetBorrowOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credentialsNotSet)
+	}
+
+	_, err := ku.GetBorrowOrder(context.Background(), "orderID")
+	if err != nil && !strings.Contains(err.Error(), "Not Found") {
+		t.Error("GetBorrowOrder() error", err)
+	}
+}
+
+func TestGetOutstandingRecord(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credentialsNotSet)
+	}
+	_, err := ku.GetOutstandingRecord(context.Background(), "BTC")
+	if err != nil {
+		t.Error("GetOutstandingRecord() error", err)
+	}
+}
+
+func TestGetRepaidRecord(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credentialsNotSet)
+	}
+	ku.Verbose = true
+	_, err := ku.GetRepaidRecord(context.Background(), "BTC")
+	if err != nil {
+		t.Error("GetRepaidRecord() error", err)
+	}
+}
+
+func TestOneClickRepayment(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test: api keys not set or canManipulateRealOrders set to false")
+	}
+
+	err := ku.OneClickRepayment(context.Background(), "BTC", "RECENTLY_EXPIRE_FIRST", 2.5)
+	if err != nil && err.Error() != "Balance insufficient" {
+		t.Error("OneClickRepayment() error", err)
+	}
+}
+
+func TestSingleOrderRepayment(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test: api keys not set or canManipulateRealOrders set to false")
+	}
+
+	err := ku.SingleOrderRepayment(context.Background(), "BTC", "fa3e34c980062c10dad74016", 2.5)
+	if err != nil && err.Error() != "Balance insufficient" {
+		t.Error("SingleOrderRepayment() error", err)
+	}
+}
+
+func TestPostLendOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test: api keys not set or canManipulateRealOrders set to false")
+	}
+
+	_, err := ku.PostLendOrder(context.Background(), "BTC", 0.001, 5, 7)
+	if err != nil && err.Error() != "Balance insufficient" {
+		t.Error("PostLendOrder() error", err)
+	}
+}
+
+func TestCancelLendOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test: api keys not set or canManipulateRealOrders set to false")
+	}
+
+	err := ku.CancelLendOrder(context.Background(), "OrderID")
+	if err != nil && err.Error() != "order not exist" {
+		t.Error("CancelLendOrder() error", err)
+	}
+}
+
+func TestSetAutoLend(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test: api keys not set or canManipulateRealOrders set to false")
+	}
+
+	err := ku.SetAutoLend(context.Background(), "BTC", 0.002, 0.005, 7, true)
```

Comment:
Please update test scenario: `400100 message: interest rates must be >= 0 and <= 0.0009`